### ISSUE TITLE
Running trials on CIFAR10 with 250,500,1000, and 2000 labels

### ISSUE THREAD

**Issue by varunnair18** [google-research/uda#58]
Hi @qizhex ,

In order to run trials on CIFAR10 on smaller labeled subsets of the data, how do we download the data and run the model appropriately?

I have downloaded the dataset using image/scripts/preprocess.sh and changing the --sup_size flag to the desired size, and then running the image/scripts/run_cifar10_gpu.sh script with the --sup_size flag also changed.

However, this did not produce the results that were listed in the paper for CIFAR10 on 250 labels, as I got results that were abnormally high.

**Comment by qizhex** [comment]
Hi, you can use scripts/preprocess.sh to generate the supervised data by setting sup_size to your desired data size and set use_equal_split to True. 

I uploaded the hyperparameters for all data sizes at scripts/run_cifar_tpu_32_core_all_sup_size.sh. For sup_size=500,1000,2000, simply setting ent_min_coeff to 0.1 works well. For sup_size=250, the hyperparameter is a bit different. You can refer to the script for more details.

**Comment by pcascanteb** [comment]
Hi @qizhex, 
After running the image/scripts/preprocess.sh, it only generates test, train and unsup tfrecords. 
Are you using the train split also as the validation split? Or are you using the standard validation set size (5k) on CIFAR10 and I'm missing something?
Thanks!

**Comment by qizhex** [comment]
Hi, to tune hyperparameters, we split part of the training as the validation. For example, when we used 4,000 examples, we used 3,200 for training and 800 for validation. After finding the best hyperparameters, we train the model on 4,000 examples without using validation and take the performance at the last step. This code has been removed in code cleaning.

**Comment by pcascanteb** [comment]
Thank you so much for your explanation!!

**Comment by jason718** [comment]
@qizhex  is `dev_size` the flag to set validation set?
https://github.com/google-research/uda/blob/d4ec238c5c358e9552b18a69a7091112db7db7c7/image/main.py#L117
